### ISSUE TITLE
How to validate every epoch and get hidden layer output in CNTK?

### ISSUE THREAD

**Issue by chhyun** [Microsoft/CNTK#891]
I'm trying CNN with CNTK and I have two questions.

I read all CNTK WIKI  and searched on the google but I couldn't find a way...

1. I want to validate with test data during training to check the test classification error and loss every 100 epochs. Is there a way to enable this ?
(training 100 epochs -> 1 validation with test data -> training 100 epochs -> 1 validation with test data ... )

2. I want to know if there is a way to get intermediate output(output of hidden layer). now I can save the network output value of test data but I need intermediate output for my research. 

Is there anyone know these methods? ( attached example code would be helpful :D )

**Comment by dongyu888** [comment]
For 1, you can just define the epoch size to be 100 X of the training set size. You then add a cvReader section to provide validation set info.

 

For 2, you can use Trace in brain script to dump information. See https://github.com/Microsoft/CNTK/wiki/Full-Function-Reference

**Comment by chhyun** [comment]
thanks that was very helpful :)

**Comment by chhyun** [comment]
May I ask one more question? How do I fix the seed value to use same weight initialization? I found my experimental results are different when I conduct same experiment with same setting. I think that's probably because of the random weight initialization. Is there a way?

**Comment by dongyu888** [comment]
No, seeds are fixed. The problem is likely caused by the fast computation inside cuDNN libraries. The order of these computations are not guaranteed and thus the final floating point computation result will be different. 

 

In the new version you can set

 

forceDeterministicAlgorithms=true

 

This will guarantee same result but will be slower.

**Comment by chhyun** [comment]
oh I get it. thanks for your answer :)
